src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -521,6 +526,11 @@ private void ReplaceVersionIdAndLastUpdatedInMeta(ResourceWrapper resourceWrappe
                 return false;
             }
 
+            if (existing.RawResource == null)
```

Comment:
NIT: Do you mind adding the reason why this condition returns true? As this method compares two inputs, and the code returns true if just one of the inputs is null, it looks strange. Probably a comment would make the reason clearer.

Reply:
This is actually not needed at all. It was leftover of first "design". Removed.